spec/models/manageiq/providers/openstack/infra_manager_spec.rb

```diff
@@ -79,4 +79,40 @@
       @ems_cloud.reload.provider.should be_nil
     end
   end
+
+  context "cloud disk usage" do
+    before do
+      @cloud = FactoryGirl.create(:ems_openstack)
+      @infra = FactoryGirl.create(:ems_openstack_infra_with_stack)
+      provider = FactoryGirl.create(:provider)
+      @cloud.provider_id = provider.id
+      @infra.provider_id = provider.id
+      @cloud.save
+      @infra.save
+    end
+
+    it "Cinder" do
+      expect(@infra.cloud_cinder_disk_usage).to eq(0)
+
+      volume = FactoryGirl.create(:cloud_volume_openstack)
+      volume.size = 11
+      @cloud.cloud_volumes << volume
+
+      expect(@infra.cloud_cinder_disk_usage).to eq(volume.size)
+    end
+
+    it "Swift" do
+      expect(@infra.cloud_swift_disk_usage).to eq(0)
+
+      container = FactoryGirl.create(:cloud_object_store_container)
+      container.bytes = 12
+      @cloud.cloud_object_store_containers << container
+
+      replicas = FactoryGirl.create(:orchestration_stack_parameter_openstack_infra_swift_replicas).value.to_i
+      object_storage_count = FactoryGirl.create(:orchestration_stack_parameter_openstack_infra_object_storage).value.to_i
+      actual_replicas = [replicas, object_storage_count].sort.first
```

Comment:
yes please, create the factories with the hardcoded values here, and compare it to them,
replicas count = 5
replicas = FactoryGirl.create(:orchestration_stack_parameter, :name = 'xxx', :value > replicas_count)
object_storage_count = FactoryGirl.create(:orchestration_stack_parameter, :name => 'yyy', :value => 20)
expect(@infra.cloud_swift_disk_usage).to eq(container.bytes * replicas_count)

and you can make another test for the second case